src/errors.ts

```diff
@@ -18,3 +18,20 @@ export function getNotExistErrorResponse(attribute: string, path: string): Promi
             variables: { path }
         }));
 }
+
+export function chromeProvidedPortWithoutUserDataDir() {
+    return Promise.reject(new ErrorWithMessage(<DebugProtocol.Message>{
+        id: 2008,
+        format: localize('random.port.no.userdatadir', 'When the remote debugging port is set to 0, you must also provide the "userDataDir" launch argument'),
+        sendTelemetry: true
+    }));
+}
+
+export function couldNotConnectToPort(address: string, port: number) {
+    return Promise.reject(new ErrorWithMessage(<DebugProtocol.Message>{
+        id: 2008,
+        format: localize('launch.port.not.open', 'Could not open a connection to Chrome at: {address}:{port}', '{address}', '{port}'),
```

Comment:
Are the latest 2 parameters necessary?

Reply:
Not sure I understand the question. You mean address and port on the localize call?

Reply:
What is the difference between:
localize('launch.port.not.open', 'Could not open a connection to Chrome at: {address}:{port}', '{address}', '{port}'),
and
localize('launch.port.not.open', 'Could not open a connection to Chrome at: {address}:{port}') ?

Reply:
Got it. Yeah I think they're just duplicated.